### ISSUE TITLE
Pixi 4 flickering on Silk browser

### ISSUE THREAD

**Issue by GopherwoodTodd** [pixijs/pixi.js#4304]
When running a pixi example in the Silk browser on a Kindle Fire the canvas flickers. We don't experience this on any other platforms or browsers.

You can see it using this example:
https://pixijs.github.io/examples/#/basics/basic.js

**Comment by ivanpopelyshev** [comment]
Ok, so there are some switches you can try:

First is "antialias". I dont know why, but it forces browsers to behave differently as a side effect.

```js
var app = new PIXI.Application(800, 600, {backgroundColor : 0x1099bb , antialias: true });
```

Second is preserveDrawingBuffer which eats the performance but affects rendering process seriously. Use with antialias.

```js
var app = new PIXI.Application(800, 600, {backgroundColor : 0x1099bb , antialias: true, preserveDrawingBuffer: true });
```

Third is calling "renderer.gl.flush" just after rendering process finished.

Either call it after "Render" if you use simple renderer
```js
renderer.render(stage);
renderer.gl.flush();
```

Either hack the app render method

```js
app = new Application(...,. { autoStart: false }) ;
app.render = function() {
    this.renderer.render(this.stage);
    this.renderer.gl.flush();
}
```

**Comment by GopherwoodTodd** [comment]
Thanks, the perserveDrawingBuffer change fixed it. Tried it without the antialias flag and it worked that way too, so only have the one.

**Comment by ivanpopelyshev** [comment]
What about single antialias? Dont forget there's performance hit with preserveDrawingBuffer. Do not enable it on other devices/browsers ;)

**Comment by GopherwoodTodd** [comment]
Antialias by itself didn't fix the problem. 

Yeah, we've limited it to just the Silk browser. It is a significant framerate hit.

**Comment by ivanpopelyshev** [comment]
Good luck! Write us more issues!

**Comment by glomotion** [comment]
Hey there,
I've also been noticing this on and off, in Chrome too. Funnily enough, adding { transparent: true } to the renderer fixes it for me.

**Comment by mafsays** [comment]
{ transparent: true } also fixes problem in Silk browser on Kindle Fire for me, without the performance impact of preserveDrawingBuffer - nice one : )

**Comment by ivanpopelyshev** [comment]
"transparent" has a performance impact but its small compared to preserveDrawingBuffer.

Thanks, @mafsays

**Comment by Tigerwulf** [comment]
Hey there,
we have the same problem, that 30% of the screen is flickering in the new Pixi-Version. We changed our version from 4.5.1 to 4.7.
Facts:
- only 30% of the renderscreen is flickering
- the bug is ONLY in Chrome on macOS, means windows or iOS runs fine

- in Chrome when you switch taps it happens that the flickering stops

I just check the commits until the flickering appears.
In version 4.5.3 was an commit "gl.SCISSOR_TEST fix (related to #4004) (#4052)" (SHA: 976c1dcdc27c1b8961db12ff66169d295d3dadfe) that enable the SCISSOR_TEST.
If i disable that line again, the flickering disappear.

I guess its not just the issue of the pixi-framework, also an problem of the core from chrome and silk or maybe the initiale data are wrong. Realy.. i dont know. But i hope we can find a solution to fix this.

For us we choose the workaround to set transparent on true, only for chrome macOS.

**Comment by guinunez** [comment]
I have been months trying to solve this problem, it was on Chrome on Huawei, and **{ transparent: true }** solved it. Thanks a lot
